libs/vm/include/vm/variant.hpp

```diff
@@ -475,37 +475,23 @@ struct Variant
   }
 };
 
-struct TemplateParameter1 : public Variant
+template <std::size_t ID = 0>
+struct VariantT : public Variant
 {
   using Variant::Variant;
+  VariantT(Variant const &other)
+    : Variant{other} {};
+  VariantT(Variant &&other)
+    : Variant{std::move(other)} {};
 };
 
-using TemplateParameter = TemplateParameter1;
-
-struct TemplateParameter2 : public Variant
-{
-  using Variant::Variant;
-};
-
-struct Any : public Variant
-{
-  using Variant::Variant;
-};
-
-struct AnyPrimitive : public Variant
-{
-  using Variant::Variant;
-};
-
-struct AnyInteger : public Variant
-{
-  using Variant::Variant;
-};
-
-struct AnyFloatingPoint : public Variant
-{
-  using Variant::Variant;
-};
+using TemplateParameter1 = VariantT<0>;
```

Comment:
This is just for a unique value from `typeid` right. Is there any downside from this approach in terms of getting the string type name? Also would it make sense to simple use `__LINE__` for the value here so that the number doesn't need to be incremented?

Or are these questions better directed at @robertdickson

Reply:
All these new types are (by design) meant to be equivalent to `Variant` type *and* at the same time being different between each other from C++ type system perspective.

Types in original implementation were lacking construction from `Variant` base class (note that original implementation used the `using Variant::Variant;` to propagate constructors but it is not enough to achieve that goal), thus it is necessary to add these constructors by hand.
This change avoids unnecessary copy-paste of per each type in original implementation.

I'm really not quite sure about using `__LINE__`.
Reason being that it somewhat masks the original intention, and makes the code dependent on formatting (e.g. in the case if someone actually wants to use `typeid(...).name` to identify the type, then these types could have different names from build to build if there was added new code somewhere above this types are declared (since it will change line numbers)).

Reply:
What's the point of this VariantT? If you want to get the official string name of a type at runtime you can ask the VM for it via VM::GetTypeInfo(type_id).name  or   GetTypeInfo(GetTypeId<T>).name

Reply:
The objective of 
The orimary objecive of the `VariantT<...>` is to implement 2 mising constructors from `Variant` (copy & move constructors).
Getting the string represention of the type (it's `name) was branched-out discussion which derailed us off the primary objective.

Reply:
But why do you need those 2 constructors? The Variant already has templated Move() and Get() calls to transfer/copy a Variant into a derived-Variant like AnyPrimitive, and such mechanisms are in use, for example, to transfer Variant's on the VM stack in to function parameters, see StackGetter < T >

Reply:
This is just to make types easily constructible from Variant. They are just empty wrappers around `Variant` base type after all.